src/crimson/os/seastore/cache.cc

```diff
@@ -121,6 +121,7 @@ void Cache::retire_extent(CachedExtentRef ref)
   assert(ref->is_valid());
 
   remove_from_dirty(ref);
+  extents.erase(*ref);
   ref->dirty_from_or_retired_at = JOURNAL_SEQ_MAX;
   retired_extent_gate.add_extent(*ref);
   ref->state = CachedExtent::extent_state_t::RETIRED;
```

Comment:
Can we set this to INVALID here?  I'm actually not sure whether we need a RETIRED state at all.

Reply:
I don't think the dirty_from_or_retired_at machinery requires the extent to be RETIRED or to be left in the extents set.

Reply:
I think you are correct that we don't need the RETIRED state in `CachedExtent`. 

The retiring should only be visable within the same transaction and we already have the logic to add it to `Transaction::retired_set`. It seems not reasonable to make the "RETIRED" state visable to other on-going transactions before submission.

As for the according transaction invalidation, it should be able to be propagated through the on-going work with the interruptable future, right?

Reply:
@athanatos replace RETIRED by INVALID, works.